libs/coin-framework/src/operation.ts

```diff
@@ -240,16 +240,67 @@ export const isAddressPoisoningOperation = (
   );
 };
 
-export function isEditableOperation(account: AccountLike, operation: Operation): boolean {
-  let isEvmFamily = false;
-  if (account.type === "Account") {
-    isEvmFamily = account.currency.family === "evm";
-  } else if (account.type === "TokenAccount") {
-    isEvmFamily = account.token.parentCurrency.family === "evm";
+export const isEditableOperation = (account: Account, operation: Operation): boolean => {
+  const { currency } = account;
+
+  // the edit transaction feature is only available for evm family
+  if (currency.family !== "evm") {
+    return false;
   }
```

Comment:
This "generated" pattern does not exist in the coin framework, and I don't think we want to add it here since coin-framework should be blockchain agnostic.
With that being said, I agree that there should not be some evm specific code / logic in here.

I updated this here a376c27ba0b0c53b3b4c886fcaece41eac42bb59